include/deal.II/base/vectorization.h

```diff
@@ -101,10 +101,19 @@ class VectorizedArrayIterator
    * @param lane A pointer to the current lane.
    */
   VectorizedArrayIterator(T &data, const unsigned int lane)
-    : data(data)
+    : data(&data)
     , lane(lane)
   {}
 
+  /**
+   * Compare for equality.
+   */
+  bool
+  operator==(const VectorizedArrayIterator<T> &other) const
+  {
+    return this->lane == other.lane;
```

Comment:
Don't you also have to check for `this->data == other.data`? Otherwise you might end up with two iterators pointing into different objects but the same lane to compare equal.

But, thinking about his more, I think the usual model is to do `Assert(this->lane == other.lane, ...)`. This is in keeping with the spirit that in C++, you are only allowed to compare iterators *into the same array*. (You are allowed to compare *pointers* across objects, but for iterators the usual restriction is that the operation is only well-defined if they point into the same object, if I recall correctly.)